student.rb

```diff
@@ -9,4 +11,9 @@ def initialize(classroom, age, parent_permission: true, name: 'Unknown')
   def play_hooky
     '¯\(ツ)/¯'
   end
+
+  def classroom=(classroom)
+    @classroom = classroom
+    classroom.add_student(self)
```

Comment:
- [ ] it would be better to add an unless exception `unless classroom.students.include?(self)` next to add_student(self) so that will ensure only add when there is no self there already